__tests__/app.test.js

```diff
@@ -105,6 +105,37 @@ describe("GET /api/articles", () => {
             }
           )
         })
+        expect(articles).toBeSortedBy('created_at', { descending: true })
+      })
+  })
+
+  test("200:responds with articles sorted by a certain property in descending order by default", () => {
+    return request(app)
+      .get('/api/articles?sort_by=votes')
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy('votes',{ descending: true });
+      })
+  })
+
+  test("200: responds with articles sorted by a certain  property in ascending order", () => {
+    return request(app)
+      .get('/api/articles?sort_by=votes&order=asc')
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy('votes',{ ascending: true });
+      })
+  })
+
+  test("200: responds with articles filtered by a chosen topic ", () => {
+    return request(app)
+      .get('/api/articles?topic=mitch')
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toHaveLength(12)
+        articles.forEach((article) => {
+          expect(article.topic).toBe('mitch')
+        })
```

Comment:
This is really good happy path testing, but one happy path test that is missing is:

When querying for a topic that exists but has no associated articles:
`/api/articles?topic=paper`


There are error handling tests that should also be included for these two tasks as well:

When given a sort_by that is invalid:
`/api/articles?sort_by=banana`

When given a order that is invalid:
`/api/articles?order=banana`

When given a topic that does not exist:
`/api/articles?topic=banana`